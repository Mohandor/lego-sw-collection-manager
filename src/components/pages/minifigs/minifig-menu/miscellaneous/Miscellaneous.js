import React from "react";
import PropTypes from "prop-types";
import { Typography, LinearProgress, Grid, Tooltip, Button, Icon } from "@material-ui/core";

export const miscellaneous = props => (
  <>
    <Grid item xs={12}>
      <Typography align="center" variant="subtitle1">
        {`You possess ${props.numberOwned} of the ${props.totalNumber} minifigs in our database`}
      </Typography>
    </Grid>
    <Grid item xs={12}>
      <Tooltip
        title={`${props.percentageOwned}%`}
        placement={props.percentageOwned > 50 ? 'top-end' : 'top'}>
        <LinearProgress
          variant="determinate"
          value={props.percentageOwned}
          className={props.classes.linearProgress}
        />
      </Tooltip>
    </Grid>
    <Grid item xs={12}>
      <Button color="primary" variant="contained" className={props.classes.button}>
        Add a minifig
        <Icon className={props.classes.icon}>add_circle</Icon>
      </Button>
      <Button
        color="primary"
        variant="contained"
        className={props.classes.button}
        href={`data:text/json;charset=utf-8,${encodeURIComponent(
          JSON.stringify(props.minifigs)
        )}`}
        download="minifigs.json"
      >
        Download minifigs
        <Icon className={props.classes.icon}>cloud_download</Icon>
      </Button>
    </Grid>
  </>
);

miscellaneous.propTypes = {
  totalNumber: PropTypes.number,
  numberOwned: PropTypes.number,
  percentageOwned: PropTypes.number,
  minifigs: PropTypes.object,
};

export default miscellaneous;
