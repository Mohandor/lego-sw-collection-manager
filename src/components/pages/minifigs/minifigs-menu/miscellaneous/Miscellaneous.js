import React from "react";
import PropTypes from "prop-types";
import {
  Typography,
  LinearProgress,
  Grid,
  Tooltip,
  Button,
  Icon,
  Divider,
  Fab
} from "@material-ui/core";

export const miscellaneous = props => (
  <>
    <Grid item xs={12}>
      <Typography align="center" variant="subtitle1">
        {`You possess ${props.numberOwned} of the ${
          props.totalNumber
        } minifigs in our database`}
      </Typography>
    </Grid>
    <Grid item xs={12}>
      <Tooltip
        title={`${props.percentageOwned}%`}
        placement={props.percentageOwned > 50 ? "top-end" : "top"}
      >
        <LinearProgress
          variant="determinate"
          value={props.percentageOwned}
          className={props.classes.linearProgress}
        />
      </Tooltip>
    </Grid>
    <Grid item xs={12}>
      <Divider />
      <Typography inline>Set possession to all:</Typography>
      <Tooltip
        title="Set all the minifigs to owned"
        placement="top"
      >
        <Fab
          size="small"
          color="primary"
          arial-label="set-all-to-possessed"
          className={props.classes.button}
          onClick={() => props.setPossessionToAll(true)}
        >
          <Icon>check_box</Icon>
        </Fab>
      </Tooltip>
      <Tooltip
        title="Set all the minifigs to missing"
        placement="top"
      >
        <Fab
          size="small"
          color="primary"
          arial-label="set-all-to-not-possessed"
          className={props.classes.button}
          onClick={() => props.setPossessionToAll(false)}
        >
          <Icon>check_box_outline_blank</Icon>
        </Fab>
      </Tooltip>
        <Divider />
    </Grid>
    <Grid item xs={12}>
      <Button
        color="primary"
        variant="contained"
        className={props.classes.button}
        disabled
      >
        Add a minifig
        <Icon className={props.classes.icon}>add_circle</Icon>
      </Button>
      <Button
        color="primary"
        variant="contained"
        className={props.classes.button}
        href={`data:text/json;charset=utf-8,${encodeURIComponent(
          JSON.stringify(props.minifigs)
        )}`}
        download="minifigs.json"
      >
        Download minifigs
        <Icon className={props.classes.icon}>cloud_download</Icon>
      </Button>
    </Grid>
  </>
);

miscellaneous.propTypes = {
  totalNumber: PropTypes.number,
  numberOwned: PropTypes.number,
  percentageOwned: PropTypes.number,
  minifigs: PropTypes.object,
  setPossessionToAll: PropTypes.func.isRequired
};

export default miscellaneous;
