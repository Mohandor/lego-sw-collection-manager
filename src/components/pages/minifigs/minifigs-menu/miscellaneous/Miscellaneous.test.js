import React from 'react';
import Miscellaneous from './Miscellaneous';
import { shallow } from 'enzyme';
import { Typography, LinearProgress, Grid, Tooltip, Button, Icon, Divider, Fab } from "@material-ui/core";

describe('<Miscellaneous />', () => {
  it('should render the miscellaneous part of the minifig menu', () => {
    const props = {
      totalNumber: 1000,
      numberOwned: 900,
      percentageOwned: 90,
      classes: {
        icon: 'icon',
        button: 'button',
        linearProgress: 'linearProgress'
      },
      setPossessionToAll: jest.fn()
    }
    const wrapper = shallow(<Miscellaneous {...props} />);
    expect(wrapper.find(Grid)).toHaveLength(4);
    expect(wrapper.find(Typography)).toHaveLength(2);
    expect(wrapper.find(LinearProgress)).toHaveLength(1);
    expect(wrapper.find(Tooltip)).toHaveLength(3);
    expect(wrapper.find(Button)).toHaveLength(2);
    expect(wrapper.find(Icon)).toHaveLength(4);
    expect(wrapper.find(Divider)).toHaveLength(2);
    expect(wrapper.find(Fab)).toHaveLength(2);
    wrapper.find(Fab).at(0).simulate('click');
    expect(props.setPossessionToAll).toHaveBeenCalledWith(true);
    wrapper.find(Fab).at(1).simulate('click');
    expect(props.setPossessionToAll).toHaveBeenCalledWith(false);
  });
});