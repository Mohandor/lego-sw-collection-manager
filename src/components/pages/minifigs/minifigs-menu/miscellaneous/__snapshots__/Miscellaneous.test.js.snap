// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Miscellaneous /> should render the miscellaneous part of the minifig menu 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Typography))
      align="center"
      variant="subtitle1"
    >
      You possess 900 of the 1000 minifigs in our database
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Tooltip))
      placement="top-end"
      title="90%"
    >
      <WithStyles(ForwardRef(LinearProgress))
        className="linearProgress"
        value={90}
        variant="determinate"
      />
    </WithStyles(ForwardRef(Tooltip))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(Typography))
      display="inline"
    >
      Set possession to all:
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Tooltip))
      placement="top"
      title="Set all the minifigs to owned"
    >
      <WithStyles(ForwardRef(Fab))
        arial-label="set-all-to-possessed"
        className="button"
        color="primary"
        onClick={[Function]}
        size="small"
      >
        <WithStyles(ForwardRef(Icon))>
          check_box
        </WithStyles(ForwardRef(Icon))>
      </WithStyles(ForwardRef(Fab))>
    </WithStyles(ForwardRef(Tooltip))>
    <WithStyles(ForwardRef(Tooltip))
      placement="top"
      title="Set all the minifigs to missing"
    >
      <WithStyles(ForwardRef(Fab))
        arial-label="set-all-to-not-possessed"
        className="button"
        color="primary"
        onClick={[Function]}
        size="small"
      >
        <WithStyles(ForwardRef(Icon))>
          check_box_outline_blank
        </WithStyles(ForwardRef(Icon))>
      </WithStyles(ForwardRef(Fab))>
    </WithStyles(ForwardRef(Tooltip))>
    <WithStyles(ForwardRef(Divider)) />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Button))
      className="button"
      color="primary"
      onClick={[MockFunction]}
      variant="contained"
    >
      Add a minifig
      <WithStyles(ForwardRef(Icon))
        className="icon"
      >
        add_circle
      </WithStyles(ForwardRef(Icon))>
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      className="button"
      color="primary"
      download="minifigs.json"
      href="data:text/json;charset=utf-8,undefined"
      variant="contained"
    >
      Download minifigs
      <WithStyles(ForwardRef(Icon))
        className="icon"
      >
        cloud_download
      </WithStyles(ForwardRef(Icon))>
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
